componenets/org.wso2.carbon.identity.application.authenticator.naver/src/main/java/org.wso2.carbon.identity.application.authenticator.naver/NaverCustomAuthenticator.java

```diff
@@ -46,6 +87,69 @@ public class NaverCustomAuthenticator extends Oauth2GenericAuthenticator {
 
     private static final long serialVersionUID = 6614257960044886319L;
 
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        return isNativeSDKBasedFederationCall(request) || super.canHandle(request);
+    }
+
+    /**
+     * Check whether the authentication is based on API.
+     *
+     * @return true since API based authentication is supported.
+     */
+    @Override
+    public boolean isAPIBasedAuthenticationSupported() {
+
+        return true;
+    }
+
+    /**
+     * Get the i18n key defined to represent the authenticator name.
+     *
+     * @return the 118n key.
+     */
+    @Override
+    public String getI18nKey() {
+
+        return NaverCustomAuthenticatorConstants.AUTHENTICATOR_I18N_KEY;
+    }
+
+    @Override
+    protected void initiateAuthenticationRequest(HttpServletRequest request, HttpServletResponse response, AuthenticationContext context) throws AuthenticationFailedException {
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("Authentication request has initialized.");
+        }
+
+        try {
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            String clientId = (String)authenticatorProperties.get(CLIENT_ID);
```

Comment:
Since authenticatorProperties is a Map<String, String>, do we need to explicitly cast it to String?

Reply:
Fixed with https://github.com/wso2-extensions/identity-outbound-auth-naver/pull/13/commits/0bd7d3f95c3c1490a22874034a254d73712cecae

Reply:
These two places haven't been fixed: https://github.com/wso2-extensions/identity-outbound-auth-naver/pull/13/files#diff-e94747e9442d81ed667af990552ceae4620d490d5953c6586dcf357090d59a6aR311-R312

Shall we fix them too?

Reply:
This id done.